src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,746 @@
+        /* We're tracing here the command, after copy in shared mem but before M0 trigger. */
+        TL_BLE_SendCmd(NULL, 0); // unused parameters for now
+      }
+      break;
+    case 2: { //ACL DATA
+        if (!acl_data_wait()) {
+#if defined(PRINT_IPCC_INFO)
+          printf("ERROR: previous ACL message not ACK'd\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+          /*  return number of bytes sent, 0 in this error case */
+          return 0;
+        }
+        TL_AclDataSerial_t *aclDataSerial = (TL_AclDataSerial_t *)(HciAclDataBuffer + sizeof(TL_PacketHeader_t));
+        aclDataSerial->type = type; // for now this param is overwritten in TL_BLE_SendCmd
+        memcpy(HciAclDataBuffer + + sizeof(TL_PacketHeader_t) + sizeof(type), pData, len);
+        TL_BLE_SendAclData(NULL, 0); // unused parameters for now
+        acl_data_on = true; /* data being send */
+      }
+      break;
+    default:
+#if defined(PRINT_IPCC_INFO)
+      printf("ERROR: not supported type\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+      while (1);
+      break;
+  }
+  return len;
+}
+
+/**
+ * Few utilities functions
+ */
+/* This function fills in a BD address table */
+static bool get_bd_address(uint8_t *bd_addr)
+{
+  uint32_t udn;
+  uint32_t company_id;
+  uint32_t device_id;
+  bool bd_found;
+
+  udn = LL_FLASH_GetUDN();
+
+  if (udn != 0xFFFFFFFF) {
+    /* "Found Unique Device Number: %#06x", udn) */
+
+    company_id = LL_FLASH_GetSTCompanyID();
+    device_id = LL_FLASH_GetDeviceID();
+
+    bd_addr[0] = (uint8_t)(udn & 0x000000FF);
+    bd_addr[1] = (uint8_t)((udn & 0x0000FF00) >> 8);
+    bd_addr[2] = (uint8_t)((udn & 0x00FF0000) >> 16);
+    bd_addr[3] = (uint8_t)device_id;
+    bd_addr[4] = (uint8_t)(company_id & 0x000000FF);
+    bd_addr[5] = (uint8_t)((company_id & 0x0000FF00) >> 8);
+
+    bd_found = true;
+  } else {
+    bd_addr = NULL;
+    bd_found = false;
+  }
+
+  return bd_found;
+}
+
+static void init_debug(void)
+{
+  /* In case of debug profile, configure debugger support */
+
+#if defined(CONFIG_DEBUG)
+#if defined(PRINT_IPCC_INFO)
+  printf("init_debug ENABLED\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+  /**
+   * Keep debugger enabled while in any low power mode
+   */
+  HAL_DBGMCU_EnableDBGSleepMode();
+  HAL_DBGMCU_EnableDBGStopMode();
+  HAL_DBGMCU_EnableDBGStandbyMode();
+
+  /***************** ENABLE DEBUGGER *************************************/
+  LL_EXTI_EnableIT_32_63(LL_EXTI_LINE_48);
+  LL_C2_EXTI_EnableIT_32_63(LL_EXTI_LINE_48);
+
+#else
+#if defined(PRINT_IPCC_INFO)
+  printf("init_debug DISABLED\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+  GPIO_InitTypeDef gpio_config = {0};
+
+  gpio_config.Pull = GPIO_NOPULL;
+  gpio_config.Mode = GPIO_MODE_ANALOG;
+
+  gpio_config.Pin = GPIO_PIN_15 | GPIO_PIN_14 | GPIO_PIN_13;
+  __HAL_RCC_GPIOA_CLK_ENABLE();
+  HAL_GPIO_Init(GPIOA, &gpio_config);
+
+  gpio_config.Pin = GPIO_PIN_4 | GPIO_PIN_3;
+  __HAL_RCC_GPIOB_CLK_ENABLE();
+  HAL_GPIO_Init(GPIOB, &gpio_config);
```

Comment:
Whythose GPIO init are required ?

Reply:
the static void init_debug(void) function coming from mbed

Reply:
OK but if user configure those pins before BLE then all his configuration will be overridden...
So there is an issue here and this does not explain why?

Reply:
this part is removed from init_debug()